### PR TITLE
RNP-104: refactor nullguarding pattern

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt [hunk 1]
```diff
@@ -209,58 +231,67 @@ class RNOneginiSdk(reactContext: ReactApplicationContext) : ReactContextBaseJava
...
                 }
-            }
-            Constants.CUSTOM_REGISTRATION_ACTION_CANCEL -> {
-                try {
-                    action.returnError(Exception(token))
-                    promise.resolve(null)
-                } catch (exception: OneginiReactNativeException) {
-                    promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
+
+                when (customAction) {
```
Comment (PiotrKlis) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt:273]: WDYT about extracting the code into smaller separate methods? 

```
 val action = registrationManager.getSimpleCustomRegistrationAction(identityProviderId)

                if (action == null) {
                    return promise.reject(OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.code, OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.message)
                }
```
and 

```
 when (customAction) {
                    Constants.CUSTOM_REGISTRATION_ACTION_PROVIDE -> {
                        try {
                            action.returnSuccess(token)
                            promise.resolve(null)
                        } catch (exception: OneginiReactNativeException) {
                            promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
                        }
                    }
                    Constants.CUSTOM_REGISTRATION_ACTION_CANCEL -> {
                        try {
                            action.returnError(Exception(token))
                            promise.resolve(null)
                        } catch (exception: OneginiReactNativeException) {
                            promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
                        }
                    }
                    else -> {
                        promise.reject(OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.code, OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.message + ". Incorrect customAction supplied: $customAction")
                    }
                }
```

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt [hunk 2]
```diff
@@ -496,37 +541,40 @@ class RNOneginiSdk(reactContext: ReactApplicationContext) : ReactContextBaseJava
     // type: User, ImplicitUser, Anonymous
```
Comment (PiotrKlis) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt:541]: comment not needed anymore

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt [hunk 3]
```diff
@@ -104,57 +106,71 @@ class RNOneginiSdk(reactContext: ReactApplicationContext) : ReactContextBaseJava
...
+            else -> sdkWrapper.getAllAuthenticators(profileId, promise)
+        }
     }
 
     @ReactMethod
     fun getRegisteredAuthenticators(profileId: String?, promise: Promise) {
-        profileId ?: promise.rejectWithNullError("profileId", "String").run { return }
-        sdkWrapper.getRegisteredAuthenticators(profileId, promise)
+        when (profileId) {
+            null -> promise.rejectWithNullError("profileId", "String")
```
Comment (PiotrKlis) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt:118]: All of the `keys` that are repeated throughout the class like `prifleId` & `pin` can be extracted into `private const`val` in `companion object` of this class.

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt (hunk 1 above)
Comment (MichielVrins) [reply android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt:273]: There is already a story to move this into a useCase, and also a story to competely rewrite this functionality. I think we can split it then if still needed.

**Review by PiotrKlis — APPROVED** [review]

**Pull request merged**